### DISCUSSION TITLE
How to convert Kekule form into aromatic form?

### DISCUSSION THREAD

**Discussion by Flora-Feng0532** [rdkit/rdkit#4441] [Q&A]
Here is my question:
if target_smiles = 'O=c1c2ccccc2ncn1'， I can get what I want: return True in main function
But if target_smiles = 'O=C1N=CNC2=C1C=CC=C2', it fails to return True all the time.
**I am greedy, I want both  'O=c1c2ccccc2ncn1' and  'O=C1N=CNC2=C1C=CC=C2' can get True in main function.**

So if I can **change kekule form('O=C1N=CNC2=C1C=CC=C2') into aromatic form'O=c1c2ccccc2ncn1' i**n the middle, problem seems to be solven. Is there any help here?

SMILES_list=['O=c1[nH]c(=O)c2cc(S(=O)(=O)Cl)ccc2[nH]1','O=C(O)Cn1cnc2ccc(N3CCN(C(=O)OCC4c5ccccc5-c5ccccc54)CC3)cc2c1=O']
'
for target_smiles in target_smiles_list:
    main(target_smiles,SMILES) # Wish it to be True in each loop
'

I would like to create query from target_smiles and getsubstructure match in main function.
Code

```
def main(target_smiles,SMILES):  
    query = create_query_option1(target_smiles)
    return Chem.MolFromSmiles(SMILES).HasSubstructMatch(query)
```

```
def create_query_option1(target_smiles):  
    scaffold = Chem.MolFromSmiles(target_smiles,sanitize=False) 
    scaffold_q = Chem.RWMol(scaffold)
    for a in scaffold_q.GetAtoms():
        a.UpdatePropertyCache()        
        if a.GetIsAromatic():
            symbol = a.GetSymbol().lower()
            smarts_atom = Chem.MolFromSmarts(f"[{symbol}]").GetAtomWithIdx(0) 
            scaffold_q.ReplaceAtom(a.GetIdx(), smarts_atom)
    return scaffold_q

```
```
def create_query_option2(target_smiles):  
    patt = Chem.MolFromSmarts(Chem.MolToSmarts(Chem.MolFromSmiles(target_smiles)))
    return patt
 
```

**Comment by greglandrum** [comment]
If I understand your question correctly, your `create_query_option2()` function looks right
Does `create_query_option2()` do what you want?

**Reply by Flora-Feng0532** [reply]
Unfortunately, Neither works
